vault/ttl.go

```diff
@@ -0,0 +1,90 @@
+package vault
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+func calculateTTL(sysView logical.SystemView, increment, backendTTL, period, backendMaxTTL, explicitMaxTTL time.Duration, startTime time.Time) (ttl time.Duration, warnings []string, errors error) {
+	// Truncate all times to the second since that is the lowest precision for
+	// TTLs
+	now := time.Now().Truncate(time.Second)
+	if startTime.IsZero() {
+		startTime = now
```

Comment:
If we're not using now elsewhere you could set startTime = `time.Now().Truncate(time.Second)` directly so save an allocation.

Reply:
Since `now` is used later, I guess this is okay.

Reply:
Yep, you can ignore this.